server/block_processor.py

```diff
@@ -566,14 +567,14 @@ def backup_blocks(self, blocks):
         coin = self.coin
         for block in blocks:
             # Check and update self.tip
-            header = coin.block_header(block, self.height)
+            header, txs = coin.block_full(block, self.height)
             header_hash = coin.header_hash(header)
```

Comment:
I'm merging your first commit of the 4 and following up with a couple of tweaks as mentioned above.  If you could rebase that would be great.  But before doing so let me make a couple of comments on the 2nd commit too.